### ISSUE TITLE
Feature Request: Autohide of dock mechanism

### ISSUE THREAD

**Issue by imikejackson** [githubuser0xFFFF/Qt-Advanced-Docking-System#635]
I like the idea of the auto hide docks and having docks being minimized on any side of the application. What are your thoughts on the ability that would allow the dock to be hidden like normal to a side of the application window, but clicking on that minimized dock put the dock back its original location before it was hidden/minimized?

For example, in CLion, this is when the "Structure" tab is hidden/minimized.
![Screenshot 2024-05-02 at 12 40 32](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/assets/5182396/1d80d71a-fac0-4f8b-bf8a-acf7efbb72b3)

This is after I click the tab on the side of the app. It always knows to go back to this original location.

![Screenshot 2024-05-02 at 12 40 46](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/assets/5182396/24d0a128-15a8-42e8-b183-7cb21b0b42cd)

Thanks for keeping this library up to date.

**Comment by githubuser0xFFFF** [comment]
What is the previous location if the user dragged various dock widgets around and closed two other dock widgets after he minimized a dock widget?

**Comment by imikejackson** [comment]
I would say what ever the last dock position a dock was in (when it is not in the minimized state) is the position that it would jump back to. In our own application, we have implemented this basically as a row of icons along the QStatusBar widget. You can toggle the visibility of a Dock widget. I would like to combine that feature with QtADS to show the dock along what ever side the user wants.

**Comment by imikejackson** [comment]
## Initial State of the app
![Screenshot 2024-05-03 at 13 05 12](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/assets/5182396/f011fac4-970b-4da9-92c7-42c413f8b24a)

## Clicking the "Pipeline Dock" 'pin' button collapses the dock to the left.
![Screenshot 2024-05-03 at 13 05 19](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/assets/5182396/07ad5ca2-41d1-4d19-82c4-898b8be2bf1d)

## Clicking that collapsed dock's button along the left side:
![Screenshot 2024-05-03 at 13 05 27](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/assets/5182396/ec63f4e1-338e-476c-b92c-4d28f24ebc13)

## Unpinning the dock does not put it back to the original location. I was hoping that I would get the original image back.
![Screenshot 2024-05-03 at 13 05 33](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/assets/5182396/f2e9ae76-004f-42ca-a275-2f239704f92f)


I am sure your design goals are different than ours. Maybe we are not selecting the proper set of config flags? not sure.

**Comment by githubuser0xFFFF** [comment]
Yes, I understand your problem. But the feature that you want only works, if the user does not change anything in the current dock layout. As soon as your user drags your bookmarks dockwidget to another location or closes it, the previous position of the Pipeline Dock is gone. Therefore I asked the question:  

What is the previous location if the user dragged various dock widgets around and closed two other dock widgets after he minimized a dock widget? The previous location maybe does not exist anymore.

Are you willing to spend some money for this feature?

**Comment by imikejackson** [comment]
Ahh.. I had to think of the scenario that you were coming up with. Now I understand the scenario. That is an interesting situation. I wonder if it does not exist, you could programmatically "recreate it"? The user could certainly recreate it through dragging. This probably isn't trivial to figure out.

Paid Support: Are you US based?

**Comment by githubuser0xFFFF** [comment]
No Germany. No, not paid support. Just to get the feature implemented. I have no intention to implement it, because I don't need it. So some money could be a motivation for me to implement this :-)

Or we just wait for a pull request from someone.

**Comment by zyrkiel** [comment]
I have opened a new issue that could be an intermediate step through the feature requested here: #672

**Comment by char101** [comment]
Isn't this doable by adding `CDockWidget.toggleViewAction` to a bottom `QToolBar`?

If you click on the button of the toggle view action, it will hide and show the widget in the same location.

**Comment by githubuser0xFFFF** [comment]
@char101 Yes, I think you are right. So this should be possible to implement in the application that uses this library.
